online_meetings/canvas_zoom_meetings.py

```diff
@@ -67,42 +72,61 @@ def get_zoom_json(self, page_num: int = 1) -> Dict:
             return zoom_json["result"]
         return None
 
+    def extract_from_js(self, key: str, text: str) -> Optional[str]:
+        """Takes a javascript text and attempts to extract a key/value (May not work with everything)
+        
+        :param key: key to extract
+        :type key: str
+        :param text: text to search in
+        :type text: str
+        :return: The matching string or None if not found
+        :rtype: Optional[str]
+        """
+        pattern = re.search(f'{key}.*"(.*)"', text)
+        if pattern:
+            return pattern.group(1)
+        return None
+
     def get_zoom_details(self, url: str, data: Dict[str, str], course_id: int):
         # Start up the zoom session
         # Initiate the LTI launch to Zoom in a session
         r = self.zoom_session.post(url=url, data=data)
+
+        # Get the scid
+        scid = self.extract_from_js("scid", r.text)
+        token = self.extract_from_js("X-XSRF-TOKEN", r.text)
+
         # Get the XSRF Token
-        pattern = re.search('"X-XSRF-TOKEN".* value:"(.*)"', r.text)
-        if pattern:
+        if token and scid:
             self.zoom_session.headers.update({
-                'X-XSRF-TOKEN': pattern.group(1)
+                'X-XSRF-TOKEN': token
             })
 
-            zoom_json = self.get_zoom_json(1)
+            zoom_json = self.get_zoom_json({'page': 1, 'lti_scid': scid})
             # The first call to zoom returns total and pageSize, get the total pages by dividing
             if zoom_json:
                 total = zoom_json["total"]
                 total_pages = math.ceil(total / zoom_json["pageSize"]) + 1
-            for page_num in range(1, total_pages):
+            for page in range(1, total_pages):
                 # Just skip the first call we've already called it, but still need to process
-                if page_num != 1:
-                    zoom_json = self.get_zoom_json(page_num)
-
-                for meeting in zoom_json["list"]:
-                    self.zoom_courses_meetings.append({
-                        'course_id': course_id,
-                        'meeting_id': meeting['meetingId'],
-                        'meeting_number': meeting['meetingNumber'],
-                        'host_id': meeting['hostId'],
-                        'topic': meeting['topic'],
-                        'join_url': meeting['joinUrl'],
-                        'start_time': meeting['startTime'],
-                        'status': meeting['status'],
-                        'timezone': meeting['timezone']
-                    })
+                if page != 1:
+                    zoom_json = self.get_zoom_json({'page': page, 'lti_scid': scid})
```

Comment:
Nitpick: I find this logic a little clunky, but it should work fine. I think in `inventory.py` I just wrote a parsing or crosswalk function so I could use it inside and outside the loop.

Reply:
Yeah, it is a little clunky. I could change it to **kwargs and maybe it'd be a little better than posting an dict of data?